### PR TITLE
[IMP] sale_order_general_discount: Apply general discount to order lines created without UI form view

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]
Bravo for including now always tests for bugfixes!

File: sale_order_general_discount/models/sale.py [hunk 1]
```diff
@@ -48,3 +48,18 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
...
+
+    @api.model
+    def create(self, vals):
+        """Apply general discount for sale order lines which are not created
+        from sale order form view.
+        """
+        if 'discount' not in vals and 'order_id' in vals:
+            sale_order = self.env['sale.order'].browse(vals['order_id'])
+            if sale_order.general_discount:
+                vals['discount'] = sale_order.general_discount
```
Comment (victor-ditay) [inline sale_order_general_discount/models/sale.py:64]: LGTM but could be done also with an default value on `discount` from `order_id.general_discount`?

**Review by victor-ditay — APPROVED** [review]

File: sale_order_general_discount/models/sale.py (hunk 1 above)
Comment (pedrobaeza) [reply sale_order_general_discount/models/sale.py:64]: In default, you don't have the `order_id` to determine the applied discount.

**Comment by rafaelbn** [comment]
Please take care of the improvements in purchase: https://github.com/OCA/purchase-workflow/pull/741

**Pull request merged**